### ISSUE TITLE
Ambiguous output from failed utests in included files

### ISSUE THREAD

**Issue by larshum** [miking-lang/miking#541]
This issue was identified by @gizemcaylak.

When a `utest` fails we only print the line at which the failed `utest` is defined in a file. This works fine when all `utest`s originate in the same file as we are running tests on. But currently, we don't remove all `utest`s from included files (I think we only remove those placed on top-level of the MLang, but I may be wrong here). This means that when an included `utest` fails, we don't know which of the included files contains this failing `utest` and we have to look this up manually. The problem applies to both `boot` and the bootstrapped evaluator and compiler.

I made a toy example, just to illustrate what the problem looks like. If we have a file `a.mc` defined as
```
include "string.mc"
let f = lam x.
  utest x with 0 in
  int2string x
```
and another file `b.mc` which includes `a.mc` as
```
include "a.mc"

utest 3 with 0
mexpr
let x = f 3 in
x
```
We get the following output when running `boot eval --test b.mc`
```
 ** Unit test FAILED on line 3 **
    LHS: 3
    RHS: 0

 ** Unit test FAILED on line 3 **
    LHS: 3
    RHS: 0
```

**Comment by dlunde** [comment]
Isn't this simply a matter of adding the file name to the printout? I believe it is stored in the info type.

**Comment by larshum** [comment]
Yeah, that's true. I'm just not sure whether we want to have included `utest`s at all, because even with the file name we still don't know where it was called from. Perhaps we should print a stack trace?

**Comment by lingmar** [comment]
I wouldn't want to remove utests from included files. I often use utests as `assert` in C, and then I want them to hit when writing the proper tests, which could be in a different file.

To get a stack trace, I typically set an `error` by the `utest` that failed, turn on `--debug-stack-trace` and run via `boot`. It's not very elegant but it's been enough for my use-cases.

**Comment by larshum** [comment]
OK, then I just think we should improve the error reporting so that one does not have to depend on `boot`.

**Comment by larshum** [comment]
I'm closing this now, since it was fixed in the now merged PR #543.
